src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c

```diff
@@ -23,26 +21,133 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
+    _this->flt164 = 20.0f;
+}
+
+void func_80B400C8(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    func_80B40308(_this, globalCtx);
+    func_80B401F8(_this, globalCtx);
+}
+
+void func_80B40100(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    s16 actorCutscene;
+    s16 actorCutscene_2;
+
+    actorCutscene = _this->initialActorCutscene;
+
+    if (ActorCutscene_GetCanPlayNext(actorCutscene) != 0) {
+        actorCutscene_2 = _this->initialActorCutscene;
+        ActorCutscene_StartAndSetUnkLinkFields(actorCutscene_2, &_this->polyActor.actor);
+
+        _this->actionFunc = &func_80B40160;
+        return;
+    }
+
+    ActorCutscene_SetIntentToPlay(_this->initialActorCutscene);
+}
+
+void func_80B40160(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    Math_StepToF(&_this->flt164, 0.0f, 0.2f);
+    _this->polyActor.actor.world.pos.y = _this->polyActor.actor.home.pos.y - _this->flt164;
+    func_80B40308(_this, globalCtx);
+
+    if (_this->flt164 <= 0.0f) {
+        ActorCutscene_Stop(_this->initialActorCutscene);
+        _this->flt164 = 0.0f;
+        func_80B40080(_this);
+    }
+
+    // SFX ID
+    func_8019F1C0(&D_80B40780, (u16)0x205EU);
+}
+
+void func_80B401F8(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    Player* player;
+    Vec3f playerWorldDistance;
+
+    if (Actor_HasParent(&_this->polyActor.actor, globalCtx) != 0) {
+        _this->polyActor.actor.parent = NULL;
+        return;
+    }
+
+    player = PLAYER;
+
+    Math_Vec3f_DistXYZAndStoreDiff(&_this->collisionHeaderMinVec3f, &player->actor.world.pos, &playerWorldDistance);
+
+    if (playerWorldDistance.x >= 0.0f && playerWorldDistance.x <= _this->collisionHeaderMaxX &&
+        playerWorldDistance.z >= 0.0f && playerWorldDistance.z <= _this->collisionHeaderMaxY &&
+        fabsf(playerWorldDistance.y) < 100.0f && player->actor.yDistToWater > 12.0f) {
+        func_800B8A1C(&_this->polyActor.actor, globalCtx, 0xBA, _this->polyActor.actor.xzDistToPlayer,
+                      fabsf(_this->polyActor.actor.yDistToPlayer));
+    }
+}
+
+void func_80B40308(BgGoronOyu* _this, GlobalContext* globalCtx) {
+    CollisionHeader* colHeader;
+    f32 sp28;
+
+    if (func_800CA1AC(globalCtx, &globalCtx->colCtx, _this->polyActor.actor.world.pos.x,
+                      _this->polyActor.actor.world.pos.z, &sp28, &colHeader) != 0) {
+        Math_Vec3s_ToVec3f(&_this->collisionHeaderMinVec3f, &colHeader->minBounds);
+        _this->collisionHeaderMaxX = colHeader->maxBounds.x;
+        _this->collisionHeaderMaxY = colHeader->maxBounds.y;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B400C8.asm")
+#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40394.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Goron_Oyu_0x80B40080/func_80B40100.asm")
+void BgGoronOyu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgGoronOyu* _this = (BgGoronOyu*)thisx;
+    void* unkPad;
```

Comment:
we tend to use something simpler for this
```suggestion
    s32 pad;
```

Reply:
This will probably become a `GlobalContext*` or whatever once we make that change anyway, so might as well make it match the rest of the repo.